test/python/scheduler/test_basic_scheduler.py

```diff
@@ -280,3 +285,123 @@ def test_barriers_in_middle(self):
+        c = ClassicalRegister(3)
+        qc = QuantumCircuit(q, c)
+        qc.measure(q[0], c[2])
+        sched = schedule(qc, backend)
+        old_sched = Schedule(
+            cmd_def.get('measure', [0, 1, 2])
+        )
+        updated_old_sched = Schedule()
+        qubit_mem_slots = {0: 2}
+        for time, inst in old_sched.instructions:
+            unused_mem_slots = [1, 0]
+            if isinstance(inst, AcquireInstruction):
+                mem_slots = []
+                for channel in inst.acquires:
+                    if channel.index in qubit_mem_slots.keys():
+                        mem_slots.append(MemorySlot(qubit_mem_slots[channel.index]))
+                    else:
+                        mem_slots.append(MemorySlot(unused_mem_slots.pop()))
+                new_acquire = AcquireInstruction(command=inst.command,
+                                                 acquires=inst.acquires,
+                                                 mem_slots=mem_slots)
+                updated_old_sched |= new_acquire << time
+            elif inst.channels[0].index in qubit_mem_slots.keys():
+                updated_old_sched |= inst << time
+        for actual, old in zip(sched.instructions, updated_old_sched.instructions):
+            self.assertEqual(actual[0], old[0])
+            self.assertEqual(actual[1].command, old[1].command)
+            self.assertEqual(actual[1].channels, old[1].channels)
+
+    def test_schedule_measure_two_in_3Q(self):
+        """Test that the new schedule for a circuit with only one qubit measured."""
+        backend = FakeOpenPulse3Q()
+        cmd_def = backend.defaults().build_cmd_def()
+        q = QuantumRegister(3)
+        c = ClassicalRegister(3)
+        qc = QuantumCircuit(q, c)
+        qc.measure(q[0], c[2])
+        qc.measure(q[1], c[1])
+        sched = schedule(qc, backend)
+        old_sched = Schedule(
+            cmd_def.get('measure', [0, 1, 2])
+        )
+        updated_old_sched = Schedule()
+        qubit_mem_slots = {0: 2, 1: 1}
+        for time, inst in old_sched.instructions:
+            unused_mem_slots = [0]
+            if isinstance(inst, AcquireInstruction):
+                mem_slots = []
+                for channel in inst.acquires:
+                    if channel.index in qubit_mem_slots.keys():
+                        mem_slots.append(MemorySlot(qubit_mem_slots[channel.index]))
+                    else:
+                        mem_slots.append(MemorySlot(unused_mem_slots.pop()))
+                new_acquire = AcquireInstruction(command=inst.command,
+                                                 acquires=inst.acquires,
+                                                 mem_slots=mem_slots)
+                updated_old_sched |= new_acquire << time
+            elif inst.channels[0].index in qubit_mem_slots.keys():
+                updated_old_sched |= inst << time
+        for actual, old in zip(sched.instructions, updated_old_sched.instructions):
```

Comment:
`self.assertEqual(sched.instructions, updated_old_sched.instructions)` does not work?

Reply:
It will work, but the whole point of this loop is to iterate through each time, command and channels of the schedule

Reply:
Do you want me to add `self.assertEqual(sched.instructions, updated_old_sched.instructions)` to each test too?